pkg/termios/termios_test.go

```diff
@@ -267,3 +269,53 @@ func TestSet(t *testing.T) {
 		}
 	}
 }
+
+// This test tries to prevent people from breaking other operating systems.
+//
+// Compare:
+//
+//	GOOS=linux   GOARCH=amd64 go doc golang.org/x/sys/unix.Termios
+//	GOOS=darwin  GOARCH=amd64 go doc golang.org/x/sys/unix.Termios
+//	GOOS=openbsd GOARCH=amd64 go doc golang.org/x/sys/unix.Termios
+//
+// It's all a mess.
+//
+// This at least makes sure that the package compiles on all platforms.
+func TestCrossCompile(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping in short mode")
+	}
+	platforms := []string{
+		"linux/386",
+		"linux/amd64",
+		"linux/arm64",
+		"darwin/amd64",
+		"darwin/arm64",
+		"freebsd/amd64",
+		"openbsd/amd64",
+	}
+	// As of Go 1.19+, running "go" selects the right one:
+	//
+	// https://go.dev/doc/go1.19
+	//
+	// "go test and go generate now place GOROOT/bin at the
+	// beginning of the PATH used for the subprocess, so tests and
+	// generators that execute the go command will resolve it to
+	// same GOROOT"
```

Comment:
Wow, that's cool. I had to fix that myself in the past... really annoying. [see e.g.](https://github.com/u-root/gobusybox/blob/46b2883a7f908fe80f7a9580e136b8d256856c47/src/pkg/golang/build.go#L127)